src/util/util.cpp

```diff
@@ -49,12 +67,18 @@ void debugf(const char* format, ...) {
 
 bool fileExists(const string& fileName)
 {
+#if defined(__cpp_lib_experimental_filesystem)
+	return std::experimental::filesystem::exists(fileName) && !std::experimental::filesystem::is_directory(fileName);
+#elif defined(__cpp_lib_filesystem) 
+	return std::filesystem::exists(fileName) && !std::filesystem::is_directory(fileName);
+#else
 	if (FILE *file = fopen(fileName.c_str(), "r"))
 	{
 		fclose(file);
 		return true;
 	}
 	return false; 
+#endif

```

Comment:
Create big code blocks is better?

Reply:
Having #ifdef blocks inside the function is fine, but I don't see the reason for adding the filesystem code here. Was something wrong with the FILE method? If so it'd better to completely replace this, ideally using something that doesn't require #ifdefs.

Reply:
@wootguy filesystem is cross platform and can work with '/' and '\\' slashes at same time, create recursive directories and etc.. But it require minimal c++14, i not remove old code for backward compatibility with c++11.

Reply:
Ok those are really useful features. Creating recursive directories is a pain to do in a cross-platform way. Still, was this added to fix a bug, or can this be added later when it's needed? I don't see `/` and `\` being a problem since you can't share configs between linux and windows.

Reply:
WINAPI can work with '/' and '\' but linux only with '/'

Reply:
I'm hesitant to use bleeding-edge features like this, especially if means older code needs to be maintained in parallel.

Linux users would have to type a \ by mistake to break something. I'm pretty sure most linux tools don't let you use \ anyway.

Reply:
only c++17 has function to convert path directly to win/linux format make_preferred(), but i test and it not working in gcc, strange :)

Reply:
For now, a "fixup" function is sufficient, i think)